src/pages/NoticeRegisterPage.tsx

```diff
@@ -1,3 +1,190 @@
+import { useState } from "react";
+
+import DatePicker from "react-datepicker";
+import { useNavigate } from "react-router-dom";
+
+import "react-datepicker/dist/react-datepicker.css";
+
+import { postNotice } from "@/apis/services/noticeService";
+import { Close } from "@/assets/icon";
+import Button from "@/components/Button";
+import TextField from "@/components/TextField";
+import { useUserStore } from "@/hooks/useUserStore";
+import { extractDigits, numberCommaFormatter } from "@/utils/number";
+
+type FormType = {
+  hourlyPay: string;
+  startsAt: Date | null;
+  workhour: string;
+  description: string;
+};
+
+const FIELD_LABELS: Record<keyof FormType, string> = {
+  hourlyPay: "시급",
+  startsAt: "시작 일시",
+  workhour: "업무 시간",
+  description: "공고 설명",
+};
+
 export default function NoticeRegisterPage() {
-  return <div>NoticeRegisterPage</div>;
+  const navigate = useNavigate();
+  const { user } = useUserStore();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [form, setForm] = useState<FormType>({
+    hourlyPay: "",
+    startsAt: null,
+    workhour: "",
+    description: "",
+  });
+
+  const handleChange = (
+    key: keyof typeof form,
+    value: string | null | Date,
+  ) => {
+    setForm((prev) => ({ ...prev, [key]: value }));
+  };
+
+  const handleSubmit = async () => {
+    if (!user?.id) {
+      alert("로그인 정보가 없습니다.");
+      return;
+    }
+
+    if (!user?.shopId) {
+      alert("가게 정보가 없습니다.");
+      return;
+    }
+
+    if (isSubmitting) return;
+
+    const requiredFields: Array<keyof FormType> = [
+      "hourlyPay",
+      "startsAt",
+      "workhour",
+    ];
+
+    const missingField = requiredFields.find((key) => {
+      const value = form[key];
+      return (
+        value === null || (typeof value === "string" && value.trim() === "")
+      );
+    });
+
+    if (missingField) {
+      alert(`${FIELD_LABELS[missingField]}을(를) 입력해 주세요.`);
+      return;
+    }
+
+    const hourlyPay = Number(extractDigits(form.hourlyPay));
+    if (isNaN(hourlyPay) || hourlyPay <= 0) {
+      alert("유효한 시급을 입력해 주세요.");
+      return;
+    }
+
+    const workhour = Number(form.workhour);
+    if (isNaN(workhour) || workhour <= 0) {
+      alert("유효한 업무 시간을 입력해 주세요.");
+      return;
+    }
+
+    try {
+      setIsSubmitting(true);
+
+      const payload = {
+        hourlyPay: hourlyPay,
+        startsAt: form.startsAt!.toISOString(),
+        workhour: workhour,
+        description: form.description.trim(),
+      };
+
+      await postNotice(user.shopId, payload);
+      navigate("/shop");
+    } finally {
+      setIsSubmitting(false);
+    }
```

Comment:
```suggestion
      setIsSubmitting(true);

      const payload = {
        hourlyPay: hourlyPay,
        startsAt: form.startsAt!.toISOString(),
        workhour: workhour,
        description: form.description.trim(),
      };

    try {
      await postNotice(user.shopId, payload);
      navigate("/shop");
    } finally {
      setIsSubmitting(false);
    }
```
실제 에러가 발생할 수 있는 구간만 try-catch 블록으로 감싸주는 게
문제 발생 가능성 구간을 더 명확하게 나타내는것 같아요 🤔